### PR TITLE
[REF] web: sample server: _mockReadGroup

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/web/static/src/views/sample_server.js [hunk 1]
```diff
@@ -673,11 +679,13 @@ export class SampleServer {
...
         const groupBy = fullGroupBy.split(":")[0];
+        const groupByField = this.data[params.model].fields[groupBy];
         const records = this.data[params.model].records;
+        const fn = getProcessFn(groupByField.type);
         for (const g of groups) {
-            const recordsInGroup = records.filter((r) => r[groupBy] === g.value);
+            const recordsInGroup = records.filter((r) => r[groupBy] === fn(g.value));
             g[`${groupBy}_count`] = recordsInGroup.length;
-            g[fullGroupBy] = [g.value, g.displayName];
+            g[fullGroupBy] = g.__serverValue;
```
Comment (Polymorphe57) [inline addons/web/static/src/views/sample_server.js:688]: invalid reconstruction

File: addons/web/static/src/views/sample_server.js [hunk 2]
```diff
@@ -436,14 +447,8 @@ export class SampleServer {
         const groups = arrayGroupBy(records, (record) => {
             const vals = {};
             for (const gb of normalizedGroupBys) {
-                const { fieldName, type } = gb;
-                let value;
-                if (["date", "datetime"].includes(type)) {
```
Comment (Polymorphe57) [inline addons/web/static/src/views/sample_server.js:441]: not useful

File: addons/web/static/src/views/sample_server.js [hunk 3]
```diff
@@ -608,9 +613,10 @@ export class SampleServer {
                     return { id: g.value, display_name: g.displayName };
                 });
             }
+            const fn = getProcessFn(groupByField.type);
             for (const r of this.data[params.model].records) {
                 const group = getSampleFromId(r.id, groups);
-                r[groupBy] = group.value;
+                r[groupBy] = fn(group.value);
```
Comment (Polymorphe57) [inline addons/web/static/src/views/sample_server.js:619]: invalid type

File: addons/web/static/src/views/sample_server.js [hunk 4]
```diff
@@ -436,14 +447,8 @@ export class SampleServer {
...
-                const { fieldName, type } = gb;
-                let value;
-                if (["date", "datetime"].includes(type)) {
-                    value = this._formatValue(record[fieldName], gb);
-                } else {
-                    value = record[fieldName];
-                }
-                vals[fieldName] = value;
+                const { fieldName } = gb;
+                vals[fieldName] = record[fieldName];
```
Comment (aab-odoo) [inline addons/web/static/src/views/sample_server.js:451]: vals[gb.fieldName] = record[gb.fieldName];

File: addons/web/static/src/views/relational_model.js [hunk 5]
```diff
@@ -2599,6 +2600,7 @@ DynamicGroupList.DEFAULT_LOAD_LIMIT = 10;
 
 export class Group extends DataPoint {
     setup(params, state) {
+        this.__serverValue = params.__serverValue;
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model.js:2603]: I think we typically use rawValue terminology in the models

File: addons/web/static/src/views/sample_server.js [hunk 6]
```diff
@@ -673,11 +679,13 @@ export class SampleServer {
         // update count and aggregates for each group
         const fullGroupBy = params.groupBy[0];
         const groupBy = fullGroupBy.split(":")[0];
+        const groupByField = this.data[params.model].fields[groupBy];
         const records = this.data[params.model].records;
+        const fn = getProcessFn(groupByField.type);
```
Comment (aab-odoo) [inline addons/web/static/src/views/sample_server.js:684]: naming could be improved. Also not sure about having a function that returns the "conversion function". Maybe simply call a function that takes a type and a value and returns the server value.

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**